### PR TITLE
Fix S2094 FP: Should not raise for messages

### LINKED ISSUE #9063: Fix S2094 FP: Should not raise for messages
### Description

It is not an uncommon pattern to have messages (commands, events) with absolutely no members in it when using messaging of any kind (Fluxor, MassTransit, Wolverine...) and especially in distributed environment. 
In such cases, rule S2094 raises.

### Repro steps

Consider the following classes, which are totally valid messages and events.

```
public class PlayCommand {}

public class StreamingServiceStartedEvent {}

```

### Expected behavior

I would expect rule S2094 not to raise on such classes.

I would suggest ignoring classes that have known suffixes such as 
- Command
- Event
- Message

### PR DESCRIPTION
**Pull request by Tim-Pohlmann**
Fixes #9063

### CODE REVIEW

File: analyzers/src/SonarAnalyzer.Common/Rules/ClassShouldNotBeEmptyBase.cs [hunk 1]
```diff
@@ -51,13 +53,17 @@ public abstract class ClassShouldNotBeEmptyBase<TSyntaxKind, TDeclarationSyntax>
...
+                    && !ShouldIgnoreBecauseOfName(identifier)
                     && !ShouldIgnoreBecauseOfBaseClassOrInterface(c.Node, c.SemanticModel))
                 {
                     c.ReportIssue(Diagnostic.Create(Rule, identifier.GetLocation(), DeclarationTypeKeyword(c.Node)));
                 }
             },
             Language.SyntaxKind.ClassAndRecordClassDeclarations);
 
+    private static bool ShouldIgnoreBecauseOfName(SyntaxToken identifier) =>
+        IgnoredSuffixes.Any(identifier.ValueText.EndsWith);
```
Comment (cristian-ambrosini-sonarsource) [inline analyzers/src/SonarAnalyzer.Common/Rules/ClassShouldNotBeEmptyBase.cs:65]: Why did you choose `.EndsWith` specifically? I guess to reduce the FN rate

File: analyzers/tests/SonarAnalyzer.Test/TestCases/ClassShouldNotBeEmpty.cs [hunk 2]
```diff
@@ -126,4 +126,8 @@ class ImplementsMarker: IMarker { }          // Compliant - implements a marker
     struct EmptyStruct { }                       // Compliant - this rule only deals with classes
 
     enum EmptyEnum { }                           // Compliant - this rule only deals with classes
+
+    class SomeCommand { }                        // Compliant, ignored because of the suffix
+    class SomeEvent { }                          // Compliant, ignored because of the suffix
+    class SomeMessage { }                        // Compliant, ignored because of the suffix
```
Comment (cristian-ambrosini-sonarsource) [inline analyzers/tests/SonarAnalyzer.Test/TestCases/ClassShouldNotBeEmpty.cs:132]: I suggest adding a few more test cases:
```cs
class SomeCommandAction { }        // Noncompliant
class SomeEventHandler { }              // Noncompliant
class SomeMessageReceived { }       // Noncompliant
```

**Review by cristian-ambrosini-sonarsource — CHANGES_REQUESTED** [review]

File: analyzers/src/SonarAnalyzer.Common/Rules/ClassShouldNotBeEmptyBase.cs (hunk 1 above)
Comment (Tim-Pohlmann) [reply analyzers/src/SonarAnalyzer.Common/Rules/ClassShouldNotBeEmptyBase.cs:65]: What else could I use here?

Comment (cristian-ambrosini-sonarsource) [reply analyzers/src/SonarAnalyzer.Common/Rules/ClassShouldNotBeEmptyBase.cs:65]: `.Contains`?

Comment (Tim-Pohlmann) [reply analyzers/src/SonarAnalyzer.Common/Rules/ClassShouldNotBeEmptyBase.cs:65]: That would not fit the ticket's specification.

File: analyzers/tests/SonarAnalyzer.Test/TestCases/ClassShouldNotBeEmpty.cs (hunk 2 above)
Comment (cristian-ambrosini-sonarsource) [reply analyzers/tests/SonarAnalyzer.Test/TestCases/ClassShouldNotBeEmpty.cs:132]: Should we check with `StringComparison.OrdinalIgnoreCase`? 
I would add at least 1 of the above UTs to test the `EndWith`, changing the implementation to `Contain` will not break any UTs as of now

**Review by cristian-ambrosini-sonarsource — CHANGES_REQUESTED** [review]

File: analyzers/tests/SonarAnalyzer.Test/TestCases/ClassShouldNotBeEmpty.cs (hunk 2 above)
Comment (Tim-Pohlmann) [reply analyzers/tests/SonarAnalyzer.Test/TestCases/ClassShouldNotBeEmpty.cs:132]: I don't think ignoring case is useful. I added UTs to document the behavior.

**Review by cristian-ambrosini-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**